DEPS

```diff
@@ -15,7 +15,14 @@ deps = {
   "vendor/bip39wally-core-native": "https://github.com/brave-intl/bip39wally-core-native.git@13bb40a215248cfbdd87d0a6b425c8397402e9e6",
   "vendor/bat-native-anonize": "https://github.com/brave-intl/bat-native-anonize.git@e3742ba3e8942eea9e4755d91532491871bd3116",
   "vendor/bat-native-tweetnacl": "https://github.com/brave-intl/bat-native-tweetnacl.git@800f9d40b7409239ff192e0be634764e747c7a75",
-  "components/brave_sync/extension/brave-sync": "https://github.com/brave/sync.git@4428f605acd31542387bfa8956dda28becaec650",
+  "components/brave_sync/extension/brave-sync": {
+    'url': 'https://github.com/brave/sync.git@4428f605acd31542387bfa8956dda28becaec650',
+    'condition': 'not checkout_android',
+  },
+  "components/brave_sync/extension/brave-sync-android": {
```

Comment:
is this right? Shouldn't it be the same directory for both, just a different commit?

Reply:
I see now, you changed the path here https://github.com/brave/brave-core/pull/4958/files#diff-4eaebbcd1dc0da11e3a3ce488290bfc6R82
which is fine I guess, but no special reason it has to be a different directory

Reply:
@bridiver , I tried to go with 
```
  "components/brave_sync/extension/brave-sync": {
    'url': 'https://github.com/brave/sync.git@4428f605acd31542387bfa8956dda28becaec650',
    'condition': 'not checkout_android',
  },
  "components/brave_sync/extension/brave-sync": {
    'url': 'https://github.com/brave/sync.git@4098493496b19f46c33a66b8867c69ee6d1a4568',
    'condition': 'checkout_android',
  },
```
but then I had
```
brave-browser/src$ gclient sync
1>Syncing projects:   0% ( 0/ 3) 

src/brave (ERROR)
----------------------------------------
[0:00:04] Started.
[0:00:04] ________ unmanaged solution; skipping src/brave
[0:00:04] Finished running: git rev-parse --verify HEAD
----------------------------------------
Traceback (most recent call last):
  File "/home/alexey/Projects/depot_tools/metrics.py", line 267, in print_notice_and_exit
    yield
  File "/home/alexey/Projects/depot_tools/gclient.py", line 3179, in <module>
    sys.exit(main(sys.argv[1:]))
  File "/home/alexey/Projects/depot_tools/gclient.py", line 3165, in main
    return dispatcher.execute(OptionParser(), argv)
  File "/home/alexey/Projects/depot_tools/subcommand.py", line 252, in execute
    return command(parser, args[1:])
  File "/home/alexey/Projects/depot_tools/gclient.py", line 2723, in CMDsync
    ret = client.RunOnDeps('update', args)
  File "/home/alexey/Projects/depot_tools/gclient.py", line 1767, in RunOnDeps
    patch_refs=patch_refs, target_branches=target_branches)
  File "/home/alexey/Projects/depot_tools/gclient_utils.py", line 937, in flush
    reraise(e[0], e[1], e[2])
  File "/home/alexey/Projects/depot_tools/gclient_utils.py", line 1014, in run
    self.item.run(*self.args, **self.kwargs)
  File "/home/alexey/Projects/depot_tools/gclient.py", line 964, in run
    self.ParseDepsFile()
  File "/home/alexey/Projects/depot_tools/gclient.py", line 724, in ParseDepsFile
    deps_content, filepath, self.get_vars(), self.get_builtin_vars())
  File "/home/alexey/Projects/depot_tools/gclient_eval.py", line 473, in Parse
    result = Exec(content, filename, vars_override, builtin_vars)
  File "/home/alexey/Projects/depot_tools/gclient_eval.py", line 379, in Exec
    value = _gclient_eval(node, filename, vars_dict)
  File "/home/alexey/Projects/depot_tools/gclient_eval.py", line 303, in _gclient_eval
    return _convert(node_or_string)
  File "/home/alexey/Projects/depot_tools/gclient_eval.py", line 259, in _convert
    key, filename, getattr(key_node, 'lineno', '<unknown>')))
ValueError: duplicate key in dictionary: components/brave_sync/extension/brave-sync (file '.../brave-browser/src/brave/DEPS', line 22)

```

line 22 is exactly the second entry for `"components/brave_sync/extension/brave-sync"` key in that case. 

So I had to use different paths.